api/envoy/service/status/v3/csds.proto

```diff
@@ -0,0 +1,97 @@
+syntax = "proto3";
+
+package envoy.service.status.v3;
+
+import "envoy/admin/v3/config_dump.proto";
+import "envoy/config/core/v3/base.proto";
+import "envoy/type/matcher/v3/node.proto";
+
+import "google/api/annotations.proto";
+import "google/protobuf/struct.proto";
+
+import "udpa/annotations/versioning.proto";
+
+option java_package = "io.envoyproxy.envoy.service.status.v3";
+option java_outer_classname = "CsdsProto";
+option java_multiple_files = true;
+option java_generic_services = true;
+
+// CSDS is Client Status Discovery Service. It can be used to get the status of
+// an xDS-compliant client from the management server's point of view. In the
+// future, it can potentially be used as an interface to get the current
+// state directly from the client.
+service ClientStatusDiscoveryService {
+  rpc StreamClientStatus(stream ClientStatusRequest) returns (stream ClientStatusResponse) {
+  }
+
+  rpc FetchClientStatus(ClientStatusRequest) returns (ClientStatusResponse) {
+    option (google.api.http).post = "/v3/discovery:client_status";
+    option (google.api.http).body = "*";
+  }
+}
+
+// Status of a config.
+enum ConfigStatus {
+  // Status info is not available/unknown.
+  UNKNOWN = 0;
+
+  // Management server has sent the config to client and received ACK.
+  SYNCED = 1;
+
+  // Config is not sent.
+  NOT_SENT = 2;
+
+  // Management server has sent the config to client but hasn’t received
+  // ACK/NACK.
+  STALE = 3;
+
+  // Management server has sent the config to client but received NACK.
+  ERROR = 4;
+}
+
+// Request for client status of clients identified by a list of NodeMatchers.
+message ClientStatusRequest {
+  option (udpa.annotations.versioning).previous_message_type =
+      "envoy.service.status.v2.ClientStatusRequest";
+
+  // Management server can use these match criteria to identify clients.
+  repeated type.matcher.v3.NodeMatcher node_matchers = 1;
```

Comment:
Do we need multiple of these if `NodeMatcher` is sufficiently expressive?

Reply:
I don't see how we could achieve requesting client status for multiple unique node ids with only a single NodeMatcher. For example, if I want to request client status for "client-abc", and "client-123", it's hard to come up with a match that will only select this two.

Reply:
So this is an OR match, right? Can you clarify this in the docs?

Reply:
Done.